core/modules/file/file.module

```diff
@@ -2789,101 +2789,68 @@ function file_get_type($file) {
 /**
  * Determines if a user may perform the given operation on the specified file.
  *
- * @param $op
+ * @param string $op
  *   The operation to be performed on the file. Possible values are:
+ *   - "create"
  *   - "view"
  *   - "download"
  *   - "update"
  *   - "delete"
- *   - "create"
- * @param $file
+ * @param object|string|NULL $file
```

Comment:
This should be `@param File|string|NULL $file`

Reply:
Looking more closely I actually don't think we should accept `NULL`. In core there's only one call to `file_access('create',  $file)` and it's got the file type. Following the pattern of other entities there's always a type/bundle to check (I think) so I don't think we want to treat File differently. From what I can tell it seems NULL was accepted before only because there were no file types until recently?

Reply:
> I actually don't think we should accept NULL

Same as above. For BC we *should* provide the ability to check without instance. And changing the param handling would be a too big change, anyway (it's a possible contrib breaker).